### PR TITLE
Remove H1-specific connection-related requirements

### PR DESCRIPTION
**Pull request by mnot**
For #144

### CODE REVIEW

**Comment by mnot** [comment]
On a case-by-case basis:

* "HTTP requirements regarding connection management" is an informative reference to h1-messaging, so it's already there;
* The requirements for 205 are just a re-statement of existing h1 message framing requirements (and duplicative); 
* The requirement for 408 is a bit odd; it's really just spelling out the implications for being in that state in h1. I don't know why it's a SHOULD.
* 413 has been refactored so it's not h1 specific, but still applicable to it.

**Comment by reschke** [comment]
Ack!

File: draft-ietf-httpbis-semantics-latest.xml [hunk 1]
```diff
@@ -767,7 +766,7 @@ Content-Type: text/plain
 <t>
    Note that the presence of an "http" or "https" URI does not imply that
    there is always an HTTP server at the identified origin listening for
-   connections. Anyone can mint a URI, whether or not a server exists and
+   TCP connections. Anyone can mint a URI, whether or not a server exists and
```
Comment (royfielding) [inline draft-ietf-httpbis-semantics-latest.xml:769]: Why was TCP added here? It's not desired.

File: draft-ietf-httpbis-semantics-latest.xml [hunk 2]
```diff
@@ -8527,8 +8511,9 @@ Content-Range: bytes 7000-7999/8000
    The <x:dfn>413 (Payload Too Large)</x:dfn> status code indicates
    that the server is refusing to process a request because the request
    payload is larger than the server is willing or able to process.
-   The server &MAY; close the connection to prevent the client from continuing
-   the request.
+   The server &MAY; terminate the request, if the protocol version in use
+   allows it; otherwise, the server &MAY; close the transport
```
Comment (royfielding) [inline draft-ietf-httpbis-semantics-latest.xml:8515]: connection, not "transport connection"

**Pull request merged**